services/user/CommonSys/common/packages/common-lib/src/messaging/supervisor.ts

```diff
@@ -47,16 +50,20 @@ const setupSupervisorIFrame = (src: string) => {
     }
 };
 
-const supervisorOrigin = siblingUrl(null, 'supervisor-sys');
+const supervisorOrigin = siblingUrl(null, "supervisor-sys");
 
+const my = new URL(window.location.href);
+const myOrigin = `${my.protocol}//${my.hostname}${my.port ? ":" + my.port : ""}`;
+
+// Convenient library for users to interact with the supervisor.
 export class Supervisor {
     public isSupervisorInitialized = false;
 
-    private pendingRequests: {
-        id: string;
+    private pendingRequest: {
+        call: FunctionCallArgs;
         resolve: (result: unknown) => void;
         reject: (result: unknown) => void;
-    }[] = [];
+    } | null = null;
```

Comment:
What's the reason to introduce a null here rather than an empty array?

Reply:
Changed from an array to a single object ("pendingRequests" to "pendingRequest"). This is because there is only one active request at a time. 

However, after talking about this specific change with John the other day the plan is to turn this back into an array in a subsequent PR and buffer the requests, submitting the next when the prior finishes.